core/src/main/java/de/jplag/TokenValueMapper.java

```diff
@@ -0,0 +1,49 @@
+package de.jplag;
+
+import java.util.HashMap;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+
+import de.jplag.logging.ProgressBarLogger;
+import de.jplag.logging.ProgressBarType;
+
+public class TokenValueMapper {
```

Comment:
This class definitely needs exhaustive documentation. First, the complexity in the GST algorithm is already high, making it harder to understand this part of the code base. Second, this class here is currently very generically named, making hard to understand what it does, why it is needed, and what is actually models conceptually.

Please add _helpful_ JDoc comments here, also include your design decision in the PR description.